File: frontend/src/models/appUser.ts
export class AppUser {
  id: string;
  accountID: string;
  password: string;
  name: string;
  appUserGroups: { id: string; name: string }[];

  constructor(
    id = "",
    accountID = "",
    password = "",
    name = "",
    appUserGroups = []
  ) {
    this.id = id;
    this.accountID = accountID;
    this.password = password;
    this.name = name;
    this.appUserGroups = appUserGroups;
  }
}

File: backend/src/app/domain/appUser.go
package domain

import "strings"

// Staff 利用者グループを示す構造体
type AppUser struct {
	ID        string    `json:"id"`
	AccountID string    `json:"accountID"`
	Password  string    `json:"password"`
	Name      string    `json:"name"`
	AppGroups AppGroups `json:"appGroups"`
}

func (u *AppUser) MatchAccountID(str string) bool {
	return u.AccountID == str
}

func (u *AppUser) MatchName(str string) bool {
	return u.Name == str
}

func (u *AppUser) LikeName(str string) bool {
	return strings.Contains(u.Name, str)
}
